runtime/near-vm-runner/src/tests/runtime_errors.rs

```diff
@@ -1043,135 +768,96 @@ mod fix_contract_loading_cost_protocol_upgrade {
-            );
-            check_failed_compilation(
-                vm_kind,
-                &near_test_contracts::LargeContract {
-                    functions: 101,
-                    locals_per_function: 9901,
-                    ..Default::default()
-                }
-                .make(),
-                CompilationError::PrepareError(PrepareError::TooManyLocals),
-            );
-            let functions_number_limit: u32 = 10_000;
-            check_failed_compilation(
-                vm_kind,
-                &near_test_contracts::LargeContract {
-                    functions: functions_number_limit / 2,
-                    panic_imports: functions_number_limit / 2 + 1,
-                    ..Default::default()
-                }
-                .make(),
-                CompilationError::PrepareError(PrepareError::TooManyFunctions),
-            );
-        });
-    }
-
-    /// For compilation tests other than checked in `test_fn_loading_gas_protocol_upgrade_fail_preparing`.
-    #[test]
-    fn test_fn_loading_gas_protocol_upgrade_fail_compiling() {
-        with_vm_variants(|vm_kind: VMKind| {
-            // This triggers `WasmerCompileError` or, for wasmtime, `LinkerError`.
-            let code = multi_memories_contract();
-            let new_version = ProtocolFeature::FixContractLoadingCost.protocol_version();
-            let old_version = new_version - 1;
-            let old_error = {
-                let (outcome, error) = make_simple_contract_call_with_protocol_version_vm(
-                    &code,
-                    "main",
-                    old_version,
-                    vm_kind,
-                );
-                assert_eq!(0, outcome.used_gas);
-                assert_eq!(0, outcome.burnt_gas);
-                error
-            };
-
-            let new_error = {
-                let (outcome, error) = make_simple_contract_call_with_protocol_version_vm(
-                    &code,
-                    "main",
-                    new_version,
-                    vm_kind,
-                );
-                let expected_gas = prepaid_loading_gas(code.len());
-                assert_eq!(expected_gas, outcome.used_gas);
-                assert_eq!(expected_gas, outcome.burnt_gas);
-                error
-            };
-
-            assert_eq!(old_error, new_error);
-        });
+        // This list covers all control flows that are expected to change
+        // with the protocol feature.
+        // Having a test for every possible preparation error would be even
+        // better, to ensure triggering any of them will always remain
+        // compatible with versions before this upgrade. Unfortunately, we
+        // currently do not have tests ready to trigger each error.
+
+        test_builder()
+            .wasm(&function_not_defined_contract())
+            .protocol_features(&[ProtocolFeature::FixContractLoadingCost])
```

Comment:
Same concern about no longer checking the old version. For all tests in `test_fn_loading_gas_protocol_upgrade_fail_preparing`.